0.1.0: Getting "Must provide query string." Response From Server

Given this GraphQL document:
```
query getAllTrips {
  allTrips {
    id
    title
    startTime
    duration
    creationTime
  }
}
```

...and this Retrofit interface:

```java
interface ServerApi {
  @POST("graphql")
  Call<Response<GetAllTrips.Data>> _allTrips(@Body GetAllTrips query);
}
```

...and this Java code to execute the query in the document:

```java
        Retrofit retrofit=new Retrofit.Builder()
          .baseUrl("https://graphql-demo.commonsware.com")
          .addConverterFactory(new ApolloConverterFactory.Builder().build())
          .addConverterFactory(MoshiConverterFactory.create())
          .build();
        ServerApi api=retrofit.create(ServerApi.class);

        try {
          retrofit2.Response<Response<GetAllTrips.Data>> response=
            api._allTrips(new GetAllTrips()).execute();
          Log.e("20170214", response.errorBody().string());
        }
        catch (IOException e) {
          Log.e("20170214", "Exception making query", e);
        }
```

...I get a 400 response from my server, and the error body of the response is:

```json
{"errors":[{"message":"Must provide query string."}]}
```

This feels like a bug, insofar as providing the query string is what the generated code is supposed to be doing, if I understand how this is all supposed to work.

The attached ZIP archive contains a complete sample project for reproducing this.

If I screwed up somewhere, I apologize.
[StaticList.zip](https://github.com/apollographql/apollo-android/files/775140/StaticList.zip)

Your version crashes, because there is no converter:

```
Process: com.commonsware.graphql.trips.simple, PID: 14180
java.lang.IllegalArgumentException: Unable to create @Body converter for class com.commonsware.graphql.trips.api.GetAllTrips (parameter #1)
    for method ServerApi._allTrips
    at retrofit2.ServiceMethod$Builder.methodError(ServiceMethod.java:720)
    at retrofit2.ServiceMethod$Builder.parameterError(ServiceMethod.java:725)
    at retrofit2.ServiceMethod$Builder.parseParameterAnnotation(ServiceMethod.java:681)
    at retrofit2.ServiceMethod$Builder.parseParameter(ServiceMethod.java:333)
    at retrofit2.ServiceMethod$Builder.build(ServiceMethod.java:202)
    at retrofit2.Retrofit.loadServiceMethod(Retrofit.java:166)
    at retrofit2.Retrofit$1.invoke(Retrofit.java:145)
    at java.lang.reflect.Proxy.invoke(Proxy.java:393)
    at com.commonsware.graphql.trips.simple.$Proxy0._allTrips(Unknown Source)
    at com.commonsware.graphql.trips.simple.MainActivity$1.run(MainActivity.java:45)
 Caused by: java.lang.IllegalArgumentException: Could not locate RequestBody converter for class com.commonsware.graphql.trips.api.GetAllTrips.
  Tried:
   * retrofit2.BuiltInConverters
   * com.apollographql.android.converter.pojo.ApolloConverterFactory
    at retrofit2.Retrofit.nextRequestBodyConverter(Retrofit.java:298)
    at retrofit2.Retrofit.requestBodyConverter(Retrofit.java:258)
    at retrofit2.ServiceMethod$Builder.parseParameterAnnotation(ServiceMethod.java:678)
      ... 7 more
```

--- a/README.md
+++ b/README.md
@@ -83,6 +83,8 @@ query DroidDetails {
 }
 ```
 
+You will also need to add your schema to the project, see instructions [HERE](http://dev.apollodata.com/ios/downloading-schema.html)
+
 From this, Apollo will generate a `DroidDetails` Java class with nested classes for reading from the network response.
 
 ```java

--- a/README.md
+++ b/README.md
@@ -9,18 +9,83 @@ Apollo-Android is a GraphQL compliant client that generates Java models from sta
 
 
 ## Usage
+Add the following to your project's top level `build.gradle` file
 
 ```groovy
-buildscript {
-  repositories {
-    mavenCentral()
-  }
-  dependencies {
-    classpath 'com.apollographql.android:gradle-plugin:$CURRENT_APOLLO_VERSION'
-  }
+
+ext {
+    kotlinVersion = '1.0.6'
+
+    compileSdkVersion = 25
+    buildToolsVersion = '25.0.2'
+    apolloVersion = '0.2.1-SNAPSHOT'
+
+    dep = [
+            androidPlugin: 'com.android.tools.build:gradle:2.3.0-beta3',
+            apolloPlugin: "com.apollographql.android:gradle-plugin:$apolloVersion",
+            apolloConverter: "com.apollographql.android:converter:$apolloVersion",
+            retrofit: 'com.squareup.retrofit2:retrofit:2.1.0',
+            kotlinGradlePlugin: "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion",
+    ]
 }
 
+subprojects {
+    buildscript {
+        repositories {
+            jcenter()
+            maven { url "https://jitpack.io" }
+            maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
+        }
+
+        dependencies {
+            classpath dep.kotlinGradlePlugin
+        }
+    }
+    repositories {
+        jcenter()
+        maven { url "https://jitpack.io" }
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
+    }
+}
+
+```
+
+within your app module's `build.gradle` add the following:
+```
+buildscript {
+    dependencies {
+        classpath dep.androidPlugin
+        classpath dep.apolloPlugin
+    }
+}
+
+apply plugin: 'com.android.application'
 apply plugin: 'com.apollographql.android'
+
+dependencies {
+    compile 'com.apollographql.android:api:0.1.0'
+    compile 'com.squareup.retrofit2:retrofit:2.1.0'
+    compile 'com.apollographql.android:converter-pojo:0.1.0'
+}
+
+android {
+    compileSdkVersion 25
+    buildToolsVersion "25.0.2"
+
+    defaultConfig {
+        minSdkVersion 15
+        targetSdkVersion 25
+    }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+}
+
+apollo {
+    generateClasses = true
+}
 ```
 
 To use Apollo, put your GraphQL queries in a `.graphql` file, like `src/main/graphql/com/example/DroidDetails.grapqhl`.  There is nothing special about this query, it can be shared with other GraphQL clients as well

--- a/README.md
+++ b/README.md
@@ -49,14 +49,15 @@ buildscript {
         classpath dep.androidPlugin
         classpath dep.apolloPlugin
     }
+
 }
 
 apply plugin: 'com.android.application'
 apply plugin: 'com.apollographql.android'
 
 dependencies {
-    compile 'com.squareup.retrofit2:retrofit:2.1.0'
-    compile 'com.apollographql.android:converter-pojo:0.1.0'
+    dep.apolloConverter
+    dep.retrofit
 }
 
 android {
@@ -67,10 +68,6 @@ android {
         minSdkVersion 15
         targetSdkVersion 25
     }
-}
-
-apollo {
-    generateClasses = true
 }
 ```
 
@@ -230,12 +227,11 @@ Retrofit retrofit = new Retrofit.Builder()
     .baseUrl(BASE_URL)
     .client(okHttpClient)
     .addConverterFactory(new ApolloConverterFactory.Builder().build())
-    .addConverterFactory(MoshiConverterFactory.create())
     .build();
 ApiService service = retrofit.create(ApiService.class);
 service.droidDetails(new OperationRequest<>(new DroidDetails()))
     application.service()
-            .droidDetails(new OperationRequest<>(new DroidDetails()))
+            .droidDetails(new new DroidDetails())
             .enqueue(new Callback<Response<DroidDetails.Data>>() {
               @Override
               public void onResponse(Call<Response<DroidDetails.Data>> call,

--- a/README.md
+++ b/README.md
@@ -14,8 +14,6 @@ Add the following to your project's top level `build.gradle` file
 ```groovy
 
 ext {
-    kotlinVersion = '1.0.6'
-
     compileSdkVersion = 25
     buildToolsVersion = '25.0.2'
     apolloVersion = '0.2.1-SNAPSHOT'
@@ -24,9 +22,7 @@ ext {
             androidPlugin: 'com.android.tools.build:gradle:2.3.0-beta3',
             apolloPlugin: "com.apollographql.android:gradle-plugin:$apolloVersion",
             apolloConverter: "com.apollographql.android:converter:$apolloVersion",
-            retrofit: 'com.squareup.retrofit2:retrofit:2.1.0',
-            kotlinGradlePlugin: "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion",
-    ]
+            retrofit: 'com.squareup.retrofit2:retrofit:2.1.0']
 }
 
 subprojects {
@@ -36,17 +32,13 @@ subprojects {
             maven { url "https://jitpack.io" }
             maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
         }
-
-        dependencies {
-            classpath dep.kotlinGradlePlugin
-        }
     }
     repositories {
         jcenter()
-        maven { url "https://jitpack.io" }
         maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
     }
 }
+
 
 ```
 
@@ -63,7 +55,6 @@ apply plugin: 'com.android.application'
 apply plugin: 'com.apollographql.android'
 
 dependencies {
-    compile 'com.apollographql.android:api:0.1.0'
     compile 'com.squareup.retrofit2:retrofit:2.1.0'
     compile 'com.apollographql.android:converter-pojo:0.1.0'
 }
@@ -75,11 +66,6 @@ android {
     defaultConfig {
         minSdkVersion 15
         targetSdkVersion 25
-    }
-
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
     }
 }
 

